src/Build/Evaluation/Expander.cs

```diff
@@ -841,7 +850,7 @@ private class MetadataMatchEvaluator
                 internal MetadataMatchEvaluator(IMetadataTable metadata, ExpanderOptions options)
                 {
                     _metadata = metadata;
-                    _options = (options & ExpanderOptions.ExpandMetadata);
```

Comment:
Why did you remove the &?

Reply:
The `_options` check lower on line 885 was missing the `Truncate` flag otherwise.

I didn't see a reason this file was doing the `&` at all, but I might have missed something.

Reply:
Yea, I don't really understand that code and whether removing that & would regress anything. The safer approach would be to preserve the truncate flag while also doing whatever the code did before, something like:

```suggestion
_options = (options & ExpanderOptions.ExpandMetadata) | (options & ExpanderOptions.Truncate) ;
```

Reply:
I made this change, I think the `&` is actually needed for this check:

https://github.com/dotnet/msbuild/blob/f365aa332f58cf43e78d76491438f15fc1988f27/src/Build/Evaluation/Expander.cs#L846